sweepbatcher/sweep_batcher.go

```diff
@@ -321,7 +330,7 @@ func (b *Batcher) handleSweep(ctx context.Context, sweep *sweep,
 	// If one of the batches accepts the sweep, we provide it to that batch.
 	for _, batch := range b.batches {
 		accepted, err := batch.addSweep(ctx, sweep)
-		if err != nil && err != ErrBatchShuttingDown {
+		if err != nil {
```

Comment:
we don't want to crash here, as it's possible that the batch confirmed by the time the sweep tries to enter it, returning an `ErrBatchShuttingDown`

Reply:
it's not critical, but no need to cause an all-around system restart 

could perhaps add some simple coverage for this, not sure how involved the unit test would be though

Reply:
In the previous loop (trying to addSweep (update) to the batch to which it already belongs), the error from `batch.addSweep` is checked with regular `err != nil` check. Should the checks of errors from `batch.addSweep` in the two loops be the same?

Another question about that loop. Can we return earlier there, if the sweep was added successfuly?

```diff
diff --git a/sweepbatcher/sweep_batcher.go b/sweepbatcher/sweep_batcher.go
index 0251101..6019f6c 100644
--- a/sweepbatcher/sweep_batcher.go
+++ b/sweepbatcher/sweep_batcher.go
@@ -388,6 +388,9 @@ func (b *Batcher) handleSweep(ctx context.Context, sweep *sweep,
                                        "accepted by batch %d", sweep.swapHash[:6],
                                        batch.id)
                        }
+
+                       // The sweep was updated in the batch, our job is done.
+                       return nil
                }
        }
 ```

Reply:
Note that there's a check below for `accepted` and if that's true we return early.

Reply:
Should this check also be `if err != nil && !errors.Is(err, ErrBatchShuttingDown) {`?

Reply:
Added the error check to both code paths.